### PR TITLE
CP-9895: Added originator for xapi login_with_password

### PR DESCRIPTION
**Pull request by koushikcgit**
The login_with_password XenAPI call takes an "originator"
string as its fourth parameter.

If a client does this, then it gets its own pool of xapi sessions.
Moreover it will not have its xapi sessions destroyed prematurely
as a result of some other misbehaving client that keeps creating
sessions and not logging out of them.

This patch adds the "originator" to all invokes of
login_with_password.

Signed-off-by: Koushik Chakravarty <koushik.chakravarty@citrix.com>

### CODE REVIEW

**Comment by xen-git** [comment]
Can one of the admins verify this patch for testing?

**Comment by thomassa** [comment]
@xen-git test this please

File: ocaml/idl/binding_sanity_checks/hannesscript.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def dictionary_list_partial_print(title, dictionary_list, keys ):
 
 #log in to the master
 session=XenAPI.Session(SERVER)
-session.login_with_password(USERNAME, PASSWORD)
+session.login_with_password(USERNAME, PASSWORD, '1.0', 'xen-api-hannesscipt.py')
```
Comment (thomassa) [inline ocaml/idl/binding_sanity_checks/hannesscript.py]: Missing letter `r`

File: ocaml/idl/ocaml_backend/python/test_client.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 # Call the server and get our result.
 print "Logging in... ",
-session = server.Session.do_login_with_password("user", "passwd")
+session = server.Session.do_login_with_password("user", "passwd", "1.0", "xen-api-test-clients.py")
```
Comment (thomassa) [inline ocaml/idl/ocaml_backend/python/test_client.py]: Should be
`...client.py"` not
`...clients.py"`

File: ocaml/idl/wire-protocol.tex [hunk 3]
```diff
@@ -175,9 +175,10 @@ \subsection{Session Layer}
 The XML-RPC interface is session-based; before you can make arbitrary RPC calls
 you must login and initiate a session. For example:
 \begin{verbatim}
-   session_id    session.login_with_password(string uname, string pwd)
+   session_id    session.login_with_password(string uname, string pwd, string version, string originator)
 \end{verbatim}
 Where {\tt uname} and {\tt password} refer to your username and password
+while {\tt version} and {\tt originator} are optional and refer to the client version and client name
```
Comment (thomassa) [inline ocaml/idl/wire-protocol.tex]: IMPORTANT: The version parameter is for the API version, not the client version.

File: scripts/examples/python/fixpbds.py [hunk 4]
```diff
@@ -56,7 +56,7 @@ def main(session,sr,map):
 
     # First acquire a valid session by logging in:
     session = XenAPI.Session(url)
-    session.xenapi.login_with_password(username, password)
+    session.xenapi.login_with_password(username, password, "1.0", "xen-api-scripts-fixpdbs.py")
```
Comment (thomassa) [inline scripts/examples/python/fixpbds.py]: Typo: transposed `b` and `d` in wrong order. Should be `fixpbds`

File: scripts/examples/python/fixpbds.py.html [hunk 5]
```diff
@@ -67,7 +67,7 @@
 
     <I><FONT COLOR="#B22222"># First acquire a valid session by logging in:
 </FONT></I>    session = XenAPI.Session(url)
-    session.xenapi.login_with_password(username, password)
+    session.xenapi.login_with_password(username, password, "1.0", "xen-api-scripts-fixpdbs.py")
```
Comment (thomassa) [inline scripts/examples/python/fixpbds.py.html]: Transposition typo:  `b` and `d` in wrong order. Should be `fixpbds`

**Comment by thomassa** [comment]
Looks good now. It just needs the commits to be squashed into one.

**Comment by thomassa** [comment]
@xen-git test this please

**Comment by thomassa** [comment]
This is good to merge as soon as it passes that testing.

**Pull request merged**